### ISSUE TITLE
error when  task ora

### ISSUE THREAD

**Issue by lg2578** [openedx-unsupported/configuration#600]
RuntimeError: maximum recursion depth exceeded

/edx/app/ora/venvs/ora/build/MySQL-python/distribute-0.6.28-py2.7.egg

Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/edx/app/ora/venvs/ora/build/MySQL-python/setup.py", line 7, in <module>

```
use_setuptools()
```

  File "distribute_setup.py", line 145, in use_setuptools

```
return _do_download(version, download_base, to_dir, download_delay)
```

  File "distribute_setup.py", line 125, in _do_download

```
_build_egg(egg, tarball, to_dir)
```

  File "distribute_setup.py", line 116, in _build_egg

```
raise IOError('Could not build the egg.')
```

IOError: Could not build the egg.

---

Cleaning up...
  Removing temporary dir /edx/app/ora/venvs/ora/build...
Command python setup.py egg_info failed with error code 1 in /edx/app/ora/venvs/ora/build/MySQL-python
Exception information:
Traceback (most recent call last):
  File "/edx/app/ora/venvs/ora/local/lib/python2.7/site-packages/pip/basecommand.py", line 122, in main
    status = self.run(options, args)
  File "/edx/app/ora/venvs/ora/local/lib/python2.7/site-packages/pip/commands/install.py", line 270, in run
    requirement_set.prepare_files(finder, force_root_egg_info=self.bundle, bundle=self.bundle)
  File "/edx/app/ora/venvs/ora/local/lib/python2.7/site-packages/pip/req.py", line 1206, in prepare_files
    req_to_install.run_egg_info()
  File "/edx/app/ora/venvs/ora/local/lib/python2.7/site-packages/pip/req.py", line 312, in run_egg_info
    command_desc='python setup.py egg_info')
  File "/edx/app/ora/venvs/ora/local/lib/python2.7/site-packages/pip/util.py", line 696, in call_subprocess
    % (command_desc, proc.returncode, cwd))
InstallationError: Command python setup.py egg_info failed with error code 1 in /edx/app/ora/venvs/ora/build/MySQL-python

**Comment by feanil** [comment]
hello, what command were you trying to run when you hit this error?

**Comment by lg2578** [comment]
just following the guid install edX on a single ubuntu 1204

**Comment by hkim823** [comment]
A lot has changed in a month, please open another issue if this is still problematic for you.

**Comment by samura** [comment]
Same here

**Comment by justinperkins1** [comment]
+1 on this, I have the same error. following directions at: https://github.com/edx/configuration/wiki/edX-Ubuntu-12.04-Installation when I issue the cmd: "wget https://raw.github.com/edx/configuration/master/util/install/vagrant.sh -O - | bash"  then it runs for a long while, and eventually fails like so: 

[code]

...(before this point, pages upon pages of script-called-script-called-script tree)...

RuntimeError: maximum recursion depth exceeded

/edx/app/discern/venvs/discern/build/MySQL-python/distribute-0.6.28-py2.7.egg

Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/edx/app/discern/venvs/discern/build/MySQL-python/setup.py", line 7, in <module>

```
use_setuptools()
```

  File "distribute_setup.py", line 145, in use_setuptools

```
return _do_download(version, download_base, to_dir, download_delay)
```

  File "distribute_setup.py", line 125, in _do_download

```
_build_egg(egg, tarball, to_dir)
```

  File "distribute_setup.py", line 116, in _build_egg

```
raise IOError('Could not build the egg.')
```

IOError: Could not build the egg.

---

Cleaning up...
Command python setup.py egg_info failed with error code 1 in /edx/app/discern/venvs/discern/build/MySQL-python
Storing debug log for failure in /edx/app/discern/.pip/pip.log

ok: [localhost] => (item=/edx/app/discern/ease/requirements.txt)

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
           to retry, use: --limit @/home/user/edx_sandbox.retry

localhost                  : ok=318  changed=72   unreachable=0    failed=1

user@EdX-dev:~$

[\code]

note: I am fairly sure this can be fixed by a judicious insertion of "sys.setrecursionlimit(2000)" or something in that ballpark. I suspect the scripts calling scripts calling scripts have grown past the default of 999 nested calls, but this isn't a looping recursion (I don't think) and is hitting a recursion defensive default of python, accidentally. (by which I mean, it isn't a loop looping around and around 999 times, it's actually many many nested scripts, I suspect)

Not sure it helps, but the os is ubuntu 12.04 freshly installed from .iso in a VMware esx 5.1 VM, and other then the apt-get update instructions, only vmtools has been installed prior to running vagrant.sh

**Comment by justinperkins1** [comment]
I have continued to try to pass this error, and so I went with the non-vagrant.sh install directions, and got the same error. The errors directed me to discern package, and so I went to "http://discern.readthedocs.org/en/latest/installation/installation_overview.html" and followed the instructions there. Results are the same. This is some progress, looks like discern's fails via python error on the cmd: "pip install -r requirements.txt"

the (very) long error is as follows: http://pastebin.com/y9p3LxYd

this is roughly the same as the error I was seeing from vagrant.sh, and so I am assuming it is the same.
